diff --git a/libps/procstat.c b/libps/procstat.c
index eac4ae4..55d5b7e 100644
--- a/libps/procstat.c
+++ b/libps/procstat.c
@@ -990,9 +990,11 @@ proc_stat_set_flags (struct proc_stat *ps, ps_flags_t flags)
     if (ps_context_find_tty_by_cttyid (ps->context, ps->cttyid, &ps->tty) == 0)
       have |= PSTAT_TTY;
 
+#if 0
   /* The number of Mach ports in the task. */
   MGET (PSTAT_NUM_PORTS, PSTAT_PID,
         proc_getnports (server, ps->pid, &ps->num_ports));
+#endif
 
   /* Update PS's flag state.  We haven't tried user flags yet, so don't mark
      them as having failed.  We do this before checking user bits so that the
