---
 opts-common.c |    2 +-
 libdiskfs/init-init.c |    3 +++
 2 file changed, 4 insertion(+), 1 deletion(-)

--- a/libdiskfs/init-init.c
+++ b/libdiskfs/init-init.c
@@ -57,6 +57,9 @@ diskfs_init_diskfs (void)
 {
   error_t err;
 
+  /* See `node-create.c'.  */
+  _diskfs_no_inherit_dir_group = 1;
+
   if (diskfs_boot_filesystem ())
     /* This is a boot filesystem, we have to do some things specially.  */
     {
--- a/libdiskfs/opts-common.c
+++ b/libdiskfs/opts-common.c
@@ -52,7 +52,7 @@ const struct argp_option diskfs_common_options[] =
   {"nogrpid",    0,   0, OPTION_ALIAS | OPTION_HIDDEN},
   {"sysvgroups", 0,   0, OPTION_ALIAS | OPTION_HIDDEN},
   {"inherit-dir-group", OPT_INHERIT_DIR_GROUP, 0, 0,
-   "Create new nodes with gid of parent dir (default)"},
+   "Create new nodes with gid of parent dir"},
   {"grpid",    0,   0, OPTION_ALIAS | OPTION_HIDDEN},
   {"bsdgroups", 0,   0, OPTION_ALIAS | OPTION_HIDDEN},
   {0, 0}
