Include procfs, random, and DDE in the build

Index: hurd-debian/Makefile
===================================================================
--- hurd-debian.orig/Makefile	2010-09-27 19:42:53.000000000 +0000
+++ hurd-debian/Makefile	2010-09-27 19:45:57.000000000 +0000
@@ -31,7 +31,8 @@
 # Hurd libraries
 lib-subdirs = libshouldbeinlibc libihash libiohelp libports libthreads \
 	      libpager libfshelp libdiskfs libtrivfs libps \
-	      libnetfs libpipe libstore libhurdbugaddr libftpconn libcons
+	      libnetfs libpipe libstore libhurdbugaddr libftpconn libcons \
+	      libmachdev libbpf libddekit libhurd-slab
 
 # Hurd programs
 prog-subdirs = auth proc exec init term \
@@ -40,7 +41,7 @@
 	       login daemons boot console \
 	       hostmux usermux ftpfs trans \
 	       console-client utils sutils ufs-fsck ufs-utils \
-	       benchmarks fstests
+	       benchmarks fstests procfs random devnode
 
 ifeq ($(HAVE_SUN_RPC),yes)
 prog-subdirs += nfs nfsd
--- hurd-debian.orig/Makeconf
+++ hurd-debian/Makeconf
@@ -548,7 +548,7 @@ vpath %.defs $(top_srcdir)/hurd
 # These we want to find in the libc include directory...
 mach_defs_names = bootstrap exc mach mach4 \
 	mach_host mach_port mach_timer_reply memory_object \
-	memory_object_default notify
+	memory_object_default notify experimental
 device_defs_names = dev_forward device device_reply device_request

 mach_defs = $(addsuffix .defs,$(mach_defs_names))
diff --git a/configure.in b/configure.in
index 05b959b..68ec159 100644
--- a/configure.in
+++ b/configure.in
@@ -227,6 +227,12 @@ AC_SUBST(boot_store_types)dnl
 AC_MSG_CHECKING(boot store types)
 AC_MSG_RESULT($boot_store_types)
 
+AC_CHECK_LIB(pciaccess, pci_system_init, LIBPCIACCESS=-lpciaccess, LIBPCIACCESS=no)
+AC_SUBST(LIBPCIACCESS)
+if test "$LIBPCIACCESS" = "no"; then
+  AC_MSG_ERROR([libpciaccess must be install in order to use libddekit])
+fi
+
 # Check for ncursesw, which is needed for the console-curses client.
 hurd_LIB_NCURSESW
 
Index: hurd-debian/procfs/Makefile
===================================================================
--- hurd-debian.orig/procfs/Makefile	2012-11-26 00:23:22.000000000 +0000
+++ hurd-debian/procfs/Makefile	2012-11-26 00:25:35.000000000 +0000
@@ -1,26 +1,31 @@
-TARGET = procfs
-OBJS = procfs.o netfs.o procfs_dir.o \
-       process.o proclist.o rootdir.o dircat.o main.o
-LIBS = -lnetfs -lps -lfshelp
+#   Makefile - for procfs
+# 
+#   Copyright (C) 2008 Free Software Foundation, Inc.
+#
+#   This program is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU General Public License as
+#   published by the Free Software Foundation; either version 2, or (at
+#   your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+dir := procfs
+makemode := server
+
+target = procfs
+
+SRCS = procfs.c netfs.c procfs_dir.c process.c proclist.c rootdir.c dircat.c main.c
+LCLHDRS = dircat.h main.h process.h procfs.h procfs_dir.h proclist.h rootdir.h
+
+OBJS = $(SRCS:.c=.o)
+HURDLIBS = netfs fshelp iohelp ps ports ihash shouldbeinlibc
+OTHERLIBS = -lpthread
 
-CC = gcc
-CFLAGS = -Wall -g
-CPPFLAGS =
-LDFLAGS =
-
-ifdef PROFILE
-CFLAGS= -g -pg
-CPPFLAGS= -DPROFILE
-LDFLAGS= -static
-LIBS= -lnetfs -lfshelp -liohelp -lps -lports -lthreads -lihash -lshouldbeinlibc
-endif
-
-CPPFLAGS += -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64
-
-all: $(TARGET)
-
-$(TARGET): $(OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
-
-clean:
-	$(RM) $(TARGET) $(OBJS)
+include ../Makeconf
