Include procfs, random, and DDE in the build

Index: hurd-debian/Makefile
===================================================================
--- hurd-debian.orig/Makefile
+++ hurd-debian/Makefile
@@ -28,7 +28,8 @@ include ./Makeconf
 # Hurd libraries
 lib-subdirs = libshouldbeinlibc libihash libiohelp libports libthreads \
 	      libpager libfshelp libdiskfs libtrivfs libps \
-	      libnetfs libpipe libstore libhurdbugaddr libftpconn libcons
+	      libnetfs libpipe libstore libhurdbugaddr libftpconn libcons \
+	      libmachdev libbpf libddekit libhurd-slab eth-filter eth-multiplexer
 
 # Hurd programs
 prog-subdirs = auth proc exec init term \
@@ -37,6 +38,7 @@ prog-subdirs = auth proc exec init term
 	       benchmarks fstests \
 	       random \
 	       procfs \
+	       devnode
 
 ifeq ($(HAVE_SUN_RPC),yes)
 prog-subdirs += nfs nfsd
Index: hurd-debian/Makeconf
===================================================================
--- hurd-debian.orig/Makeconf
+++ hurd-debian/Makeconf
@@ -576,7 +576,7 @@ vpath %.defs $(top_srcdir)/hurd
 # These we want to find in the libc include directory...
 mach_defs_names = bootstrap exc mach mach4 \
 	mach_host mach_port mach_timer_reply memory_object \
-	memory_object_default notify
+	memory_object_default notify experimental
 mach_debug_defs_names = mach_debug
 device_defs_names = dev_forward device device_reply device_request
 
Index: hurd-debian/configure.ac
===================================================================
--- hurd-debian.orig/configure.ac
+++ hurd-debian/configure.ac
@@ -241,6 +241,12 @@ AC_SUBST(boot_store_types)dnl
 AC_MSG_CHECKING(boot store types)
 AC_MSG_RESULT($boot_store_types)
 
+AC_CHECK_LIB(pciaccess, pci_system_init, LIBPCIACCESS=-lpciaccess, LIBPCIACCESS=no)
+AC_SUBST(LIBPCIACCESS)
+if test "$LIBPCIACCESS" = "no"; then
+  AC_MSG_ERROR([libpciaccess must be install in order to use libddekit])
+fi
+
 # Check for ncursesw, which is needed for the console-curses client.
 hurd_LIB_NCURSESW
 
