Include DDE in the build

--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,12 @@ include ./Makeconf
 # Hurd libraries
 lib-subdirs = libshouldbeinlibc libihash libiohelp libports libthreads \
 	      libpager libfshelp libdiskfs libtrivfs libps \
-	      libnetfs libpipe libstore libhurdbugaddr libftpconn libcons
+	      libnetfs libpipe libstore libhurdbugaddr libftpconn libcons \
+	      libbpf libhurd-slab eth-filter eth-multiplexer
+
+ifneq ($(LIBPCIACCESS),no)
+lib-subdirs += libmachdev libddekit
+endif
 
 # Hurd programs
 prog-subdirs = auth proc exec term \
@@ -42,6 +47,7 @@ prog-subdirs = auth proc exec term \
 	       procfs \
 	       startup \
 	       init \
+	       devnode
 
 ifeq ($(HAVE_SUN_RPC),yes)
 prog-subdirs += nfs nfsd
--- a/Makeconf
+++ b/Makeconf
@@ -576,7 +576,7 @@ vpath %.defs $(top_srcdir)/hurd
 # These we want to find in the libc include directory...
 mach_defs_names = bootstrap exc mach mach4 \
 	mach_host mach_port mach_timer_reply memory_object \
-	memory_object_default notify \
+	memory_object_default notify experimental \
 	gnumach \
 	task_notify \
 
--- a/configure.ac
+++ b/configure.ac
@@ -245,6 +245,12 @@ AC_SUBST(boot_store_types)dnl
 AC_MSG_CHECKING(boot store types)
 AC_MSG_RESULT($boot_store_types)
 
+AC_CHECK_LIB(pciaccess, pci_system_init, LIBPCIACCESS=-lpciaccess, LIBPCIACCESS=no)
+AC_SUBST(LIBPCIACCESS)
+if test "$LIBPCIACCESS" = "no"; then
+  AC_MSG_WARN([libpciaccess must be installed in order to use libddekit])
+fi
+
 # Check for ncursesw, which is needed for the console-curses client.
 hurd_LIB_NCURSESW
 
--- a/config.make.in
+++ b/config.make.in
@@ -98,6 +98,9 @@ libblkid_LIBS = @libblkid_LIBS@
 # Whether Sun RPC support is available.
 HAVE_SUN_RPC = @HAVE_SUN_RPC@
 
+# Whether libpciaccess is available.
+LIBPCIACCESS = @LIBPCIACCESS@
+
 # Installation tools.
 INSTALL = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
