Index: hurd-debian/Makeconf
===================================================================
--- hurd-debian.orig/Makeconf
+++ hurd-debian/Makeconf
@@ -77,7 +77,7 @@ INCLUDES += -I.. $(top_srcdirinc)
 endif
 INCLUDES += -I$(..)include -I$(top_srcdir)/include
 CPPFLAGS += $(INCLUDES) \
-	     -D_GNU_SOURCE -D_IO_MTSAFE_IO -D_FILE_OFFSET_BITS=64 \
+	     -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 \
 	    $($*-CPPFLAGS)
 CFLAGS += -std=gnu99 $(gnu89-inline-CFLAGS) -Wall -g -O3 \
 	    $($*-CFLAGS)
Index: hurd-debian/libthreads/lockfile.c
===================================================================
--- hurd-debian.orig/libthreads/lockfile.c
+++ hurd-debian/libthreads/lockfile.c
@@ -17,6 +17,9 @@
    write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
    Boston, MA 02111-1307, USA.  */
 
+#define _IO_MTSAFE_IO
+#define IS_IN(lib) 0
+
 #include <cthreads.h>		/* Must come before <stdio.h>! */
 #include <stdio.h>
 
