libexec is not FHS

Index: hurd-debian/config/ttys
===================================================================
--- hurd-debian.orig/config/ttys
+++ hurd-debian/config/ttys
@@ -4,11 +4,11 @@
 
 # name	program				type		status	comments
 
-console	"/libexec/getty 9600"		mach-gnu-color	on	secure trusted console
-tty1	"/libexec/getty 38400"		hurd		on	secure trusted console
-tty2	"/libexec/getty 38400"		hurd		on	secure trusted console
-tty3	"/libexec/getty 38400"		hurd		on	secure trusted console
-tty4	"/libexec/getty 38400"		hurd		on	secure trusted console
-tty5	"/libexec/getty 38400"		hurd		on	secure trusted console
-tty6	"/libexec/getty 38400"		hurd		on	secure trusted console
-#com0	"/libexec/getty 9600"		dialup		on	secure
+console	"/sbin/getty 9600"		mach-gnu-color	on	secure trusted console
+tty1	"/sbin/getty 38400"		hurd		on	secure trusted console
+tty2	"/sbin/getty 38400"		hurd		on	secure trusted console
+tty3	"/sbin/getty 38400"		hurd		on	secure trusted console
+tty4	"/sbin/getty 38400"		hurd		on	secure trusted console
+tty5	"/sbin/getty 38400"		hurd		on	secure trusted console
+tty6	"/sbin/getty 38400"		hurd		on	secure trusted console
+#com0	"/sbin/getty 9600"		dialup		on	secure
Index: hurd-debian/doc/hurd.texi
===================================================================
--- hurd-debian.orig/doc/hurd.texi
+++ hurd-debian/doc/hurd.texi
@@ -566,7 +566,7 @@ The @option{--multiboot-command-line} op
 it is a root filesystem, which triggers it to run @command{/hurd/startup} as PID
 2.  @command{/hurd/startup} starts the @command{/hurd/proc} and
 @command{/hurd/auth} servers.  After the servers are launched
-@command{/hurd/startup} starts the @command{/libexec/runsystem.sh} script to
+@command{/hurd/startup} starts the @command{/etc/hurd/runsystem.sh} script to
 finish booting.
 
 After the Hurd has been booted, other sets of core Hurd servers can be
Index: hurd-debian/startup/startup.c
===================================================================
--- hurd-debian.orig/startup/startup.c
+++ hurd-debian/startup/startup.c
@@ -912,7 +912,7 @@ frob_kernel_process (void)
 /** Running userland.  **/
 
 /* In the "split-init" setup, we just run a single program (usually
-   /libexec/runsystem) that is not expected to ever exit (or stop).
+   /etc/hurd/runsystem) that is not expected to ever exit (or stop).
    If it does exit (or can't be started), we go to an emergency single-user
    shell as a fallback.  */
 
@@ -980,7 +980,7 @@ process_signal (int signo)
     }
 }
 
-/* Start the child program PROG.  It is run via /libexec/console-run
+/* Start the child program PROG.  It is run via /sbin/console-run
    with the given additional arguments.  */
 static int
 start_child (const char *prog, char **progargs)
@@ -992,7 +992,7 @@ start_child (const char *prog, char **pr
 
   if (progargs == 0)
     {
-      const char *argv[] = { "/libexec/console-run", prog, 0 };
+      const char *argv[] = { "/sbin/console-run", prog, 0 };
       err = argz_create ((char **) argv, &args, &arglen);
     }
   else
@@ -1002,7 +1002,7 @@ start_child (const char *prog, char **pr
 	++argc;
       {
 	const char *argv[2 + argc + 1];
-	argv[0] = "/libexec/console-run";
+	argv[0] = "/sbin/console-run";
 	argv[1] = prog;
 	argv[2 + argc] = 0;
 	while (argc-- > 0)
@@ -1062,8 +1062,8 @@ launch_something (const char *why)
   static unsigned int try;
   static const char *const tries[] =
   {
-    "/libexec/runsystem",
-    "/libexec/runsystem.gnu",
+    "/etc/hurd/runsystem",
+    "/etc/hurd/runsystem.gnu",
     _PATH_BSHELL,
     "/bin/shd",			/* XXX */
   };
