Index: libpthread/include/pthread/pthread.h
===================================================================
RCS file: /cvsroot/hurd/hurd/libpthread/include/pthread/pthread.h,v
retrieving revision 1.15
retrieving revision 1.16
diff -u -p -r1.15 -r1.16
--- libpthread/include/pthread/pthread.h	12 Aug 2008 15:07:49 -0000	1.15
+++ libpthread/include/pthread/pthread.h	21 Dec 2008 18:41:27 -0000	1.16
@@ -521,6 +521,7 @@ extern int pthread_rwlockattr_setpshared
 
 #include <bits/rwlock.h>
 
+#define PTHREAD_RWLOCK_INITIALIZER __PTHREAD_RWLOCK_INITIALIZER
 /* Create a rwlock object with attributes given by ATTR and strore the
    result in *RWLOCK.  */
 extern int pthread_rwlock_init (pthread_rwlock_t *__restrict rwlock,
