Index: libpthread/include/pthread/pthread.h
===================================================================
RCS file: /cvsroot/hurd/hurd/libpthread/include/pthread/pthread.h,v
retrieving revision 1.17
diff -u -p -r1.17 pthread.h
--- libpthread/include/pthread/pthread.h	10 Jan 2009 13:22:26 -0000	1.17
+++ libpthread/include/pthread/pthread.h	8 May 2009 00:14:04 -0000
@@ -310,12 +310,6 @@ extern int pthread_mutexattr_settype(pth
 #include <bits/mutex.h>
 
 #define PTHREAD_MUTEX_INITIALIZER __PTHREAD_MUTEX_INITIALIZER
-/* Static initializer for recursive mutexes.  */
-
-#ifdef __USE_GNU
-# define PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP \
-  __PTHREAD_RECURSIVE_MUTEX_INITIALIZER
-#endif
 
 /* Create a mutex with attributes given by ATTR and store it in
    *__MUTEX.  */
