diff --git a/Makefile b/Makefile
index 3194473..585d13a 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,8 @@ DIST_FILES = COPYING Makeconf config.make.in configure.in configure \
 # Hurd libraries
 lib-subdirs = libshouldbeinlibc libihash libiohelp libports libthreads \
 	      libpager libfshelp libdiskfs libtrivfs libps \
-	      libnetfs libpipe libstore libhurdbugaddr libftpconn libcons
+	      libnetfs libpipe libstore libhurdbugaddr libftpconn libcons \
+	      libpthread
 
 # Hurd programs
 prog-subdirs = auth proc exec init term \
@@ -40,7 +41,7 @@ prog-subdirs = auth proc exec init term \
 	       login daemons nfsd boot console \
 	       hostmux usermux ftpfs trans \
 	       console-client utils sutils ufs-fsck ufs-utils \
-	       benchmarks fstests
+	       benchmarks fstests procfs
 
 # Other directories
 other-subdirs = hurd doc config release include
