Index: libpthread/sysdeps/generic/bits/mutex.h
===================================================================
RCS file: /cvsroot/hurd/hurd/libpthread/sysdeps/generic/bits/mutex.h,v
retrieving revision 1.3
diff -u -p -r1.3 mutex.h
--- libpthread/sysdeps/generic/bits/mutex.h	12 May 2005 20:55:38 -0000	1.3
+++ libpthread/sysdeps/generic/bits/mutex.h	10 Dec 2005 19:45:49 -0000
@@ -75,7 +75,8 @@ pthread_mutex_init (struct __pthread_mut
   struct __pthread_mutex initialized_mutex = __PTHREAD_MUTEX_INITIALIZER;
 
   extern int _pthread_mutex_init (struct __pthread_mutex *,
-				  const pthread_mutexattr_t *);
+				  const pthread_mutexattr_t *)
+    __attribute__ ((weak));
 
   if (attr)
     return _pthread_mutex_init (__mutex, attr);
@@ -87,7 +88,8 @@ pthread_mutex_init (struct __pthread_mut
 _EXTERN_INLINE int
 pthread_mutex_destroy (struct __pthread_mutex *__mutex)
 {
-  extern int _pthread_mutex_destroy (struct __pthread_mutex *);
+  extern int _pthread_mutex_destroy (struct __pthread_mutex *)
+    __attribute__ ((weak));
 
   if (__mutex->attr || __mutex->data)
     return _pthread_mutex_destroy (__mutex);
@@ -98,7 +100,8 @@ pthread_mutex_destroy (struct __pthread_
 _EXTERN_INLINE int
 __pthread_mutex_lock (struct __pthread_mutex *__mutex)
 {
-  extern int _pthread_mutex_lock (struct __pthread_mutex *);
+  extern int _pthread_mutex_lock (struct __pthread_mutex *)
+    __attribute__ ((weak));
   
   if (__mutex->attr == NULL
       && __mutex->data == NULL
@@ -111,7 +114,8 @@ __pthread_mutex_lock (struct __pthread_m
 extern __inline int
 __pthread_mutex_trylock (struct __pthread_mutex *__mutex)
 {
-  extern int _pthread_mutex_trylock (struct __pthread_mutex *);
+  extern int _pthread_mutex_trylock (struct __pthread_mutex *)
+    __attribute__ ((weak));
   
   if (__mutex->attr == NULL
       && __mutex->data == NULL)
