---
 libpthread/sysdeps/generic/pt-mutex-trylock.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/libpthread/sysdeps/generic/pt-mutex-trylock.c
+++ b/libpthread/sysdeps/generic/pt-mutex-trylock.c
@@ -34,6 +34,7 @@ __pthread_mutex_trylock (struct __pthrea
   if (__pthread_spin_trylock (&mutex->__held) == 0)
     /* Acquired the lock.  */
     {
+#if defined(ALWAYS_TRACK_MUTEX_OWNER)
 #ifndef NDEBUG
       self = _pthread_self ();
       if (self)
@@ -45,6 +46,7 @@ __pthread_mutex_trylock (struct __pthrea
 	  mutex->owner = _pthread_self ();
 	}
 #endif
+#endif
 
       if (mutex->attr)
 	switch (mutex->attr->mutex_type)
