Just a build fix.

# missing mach_notify_new_task RPCs

To be removed with glibc 2.21

Index: hurd-debian/proc/Makefile
===================================================================
--- hurd-debian.orig/proc/Makefile
+++ hurd-debian/proc/Makefile
@@ -28,7 +28,7 @@ MIGSFLAGS = -imacros $(srcdir)/mig-mutat
 
 MIGSTUBS = processServer.o notifyServer.o \
 	ourmsgUser.o proc_excUser.o proc_excServer.o \
-	task_notifyServer.o
+	task_notifyServer.o task_notifyUser.o
 OBJS = $(SRCS:.c=.o) $(MIGSTUBS)
 HURDLIBS = ihash ports shouldbeinlibc
 LDLIBS = -lpthread
