Index: hurd/daemons/runsystem.sh
===================================================================
--- hurd.orig/daemons/runsystem.sh	2011-12-06 01:30:23.371425299 +0100
+++ hurd/daemons/runsystem.sh	2011-12-06 01:31:42.029647704 +0100
@@ -61,6 +61,11 @@ if ! test -e /servers/socket/1 && which
   settrans -c /servers/socket/1 /hurd/pflocal
 fi
 
+# until we properly start /etc/rcS.d
+rm -fr /run/*
+mkdir -p /run/lock /run/shm
+chmod 1777 /run/lock /run/shm
+
 # We expect to be started by console-run, which gives us no arguments and
 # puts FALLBACK_CONSOLE=file-name in the environment if our console is
 # other than a normal /dev/console.
