--- release/servers.boot	1999-05-16 01:16:43.000000000 +0200
+++ release/servers.boot.new	2007-07-31 15:39:02.000000000 +0200
@@ -3,7 +3,8 @@
 
 # First, the bootstrap filesystem.  It needs several ports as arguments,
 # as well as the user flags from the boot loader.
-/hurd/ufs.static --bootflags=${boot-args} --host-priv-port=${host-port} --device-master-port=${device-port} --exec-server-task=${exec-task} -Tdevice ${root-device} $(task-create) $(task-resume)
+/hurd/ext2fs.static --multiboot-command-line=${kernel-command-line} --host-priv-port=${host-port} --device-master-port=${device-port} --exec-server-task=${exec-task} -T typed ${root} $(task-create) $(task-resume)
+
 
 # Now the exec server; to load the dynamically-linked exec server program,
 # we have the boot loader in fact load and run ld.so, which in turn
@@ -11,6 +12,3 @@
 # in ${exec-task} to be passed to the fs above, but it is left suspended;
 # the fs will resume the exec task once it is ready.
 /lib/ld.so.1 /hurd/exec $(exec-task=task-create)
-
-# default pager
-/dev/sd0b $(add-paging-file)
