Index: libdiskfs/file-utimes.c
===================================================================
RCS file: /cvsroot/hurd/hurd/libdiskfs/file-utimes.c,v
retrieving revision 1.9
diff -u -r1.9 file-utimes.c
--- libdiskfs/file-utimes.c	2 Feb 2009 03:04:46 -0000	1.9
+++ libdiskfs/file-utimes.c	4 Apr 2009 22:47:50 -0000
@@ -33,7 +33,7 @@
 			 else
 			   {
 			     np->dn_stat.st_atim.tv_sec = atime.seconds;
-			     np->dn_stat.st_atim.tv_nsec = atime.microseconds * 1000;
+			     np->dn_stat.st_atim.tv_nsec = 0;
 			     np->dn_set_atime = 0;
 			   }
 			 
@@ -42,7 +42,7 @@
 			 else
 			   {
 			     np->dn_stat.st_mtim.tv_sec = mtime.seconds;
-			     np->dn_stat.st_mtim.tv_nsec = mtime.microseconds * 1000;
+			     np->dn_stat.st_mtim.tv_nsec = 0;
 			     np->dn_set_mtime = 0;
 			   }
 			 
Index: libdiskfs/node-times.c
===================================================================
RCS file: /cvsroot/hurd/hurd/libdiskfs/node-times.c,v
retrieving revision 1.15
diff -u -r1.15 node-times.c
--- libdiskfs/node-times.c	16 Feb 2009 16:06:00 -0000	1.15
+++ libdiskfs/node-times.c	4 Apr 2009 22:47:51 -0000
@@ -53,21 +53,21 @@
   if (np->dn_set_mtime)
     {
       np->dn_stat.st_mtim.tv_sec = t.tv_sec;
-      np->dn_stat.st_mtim.tv_nsec = t.tv_usec * 1000;
+      np->dn_stat.st_mtim.tv_nsec = 0;
       np->dn_stat_dirty = 1;
       np->dn_set_mtime = 0;
     }
   if (np->dn_set_atime)
     {
       np->dn_stat.st_atim.tv_sec = t.tv_sec;
-      np->dn_stat.st_atim.tv_nsec = t.tv_usec * 1000;
+      np->dn_stat.st_atim.tv_nsec = 0;
       np->dn_stat_dirty = 1;
       np->dn_set_atime = 0;
     }
   if (np->dn_set_ctime)
     {
       np->dn_stat.st_ctim.tv_sec = t.tv_sec;
-      np->dn_stat.st_ctim.tv_nsec = t.tv_usec * 1000;
+      np->dn_stat.st_ctim.tv_nsec = 0;
       np->dn_stat_dirty = 1;
       np->dn_set_ctime = 0;
     }
